crates/bevy_sprite/src/mesh2d/material.rs

```diff
@@ -175,6 +176,26 @@ where
     }
 }
 
+fn extract_material_meshes_2d<M: Material2d>(
+    mut commands: Commands,
+    mut previous_len: Local<usize>,
+    query: Extract<Query<(Entity, &ViewVisibility, &Handle<M>)>>,
+) {
+    let mut values = Vec::with_capacity(*previous_len);
+    for (entity, view_visibility, material) in &query {
+        if view_visibility.get() {
+            // NOTE: Material2dBindGroupId is inserted here to avoid a table move. Upcoming changes
+            // to use SparseSet for render world entity storage will do this automatically.
+            values.push((
+                entity,
+                (material.clone_weak(), Material2dBindGroupId::default()),
```

Comment:
With Bevy Asset V2, generally if you're storing weak `Handle<T>`, you should instead be storing `AssetId<T>`. You would instead call `material.id()` here to statically state that this is _not_ an owned handle (instead of making it implicit). This will also help us (eventually) move to a strong-only handle system.

Reply:
Ah, I thought I noticed that hash maps were now keyed on id instead of the handle. I'll have a look through at this and change to using ids everywhere instead if possible.

Reply:
AssetId isn't a component?

Reply:
I'll leave this until another PR as well as it's a bit more involved than meets the eye and isn't part of batching.

Reply:
Yup this cross cuts more than I thought. Sounds good!